src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -22,6 +23,90 @@ namespace System
     /// </summary>
     public static class Span
     {
+        /// <summary>
+        /// Determines whether the beginning of the span matches the specified value when compared using the specified comparison option.
+        /// </summary>
+        public static bool StartsWith(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparison)
+        {
+            if ((uint)(comparison - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
+            {
+                throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparison));
+            }
+
+            if (span.Length < value.Length)
+            {
+                return false;
+            }
```

Comment:
Is this length check valid for all cultures? cc: @tarekgh

Reply:
No, this check is wrong for culture aware cases

---
In reply to: [166252147](https://github.com/dotnet/coreclr/pull/16223#discussion_r166252147) [](ancestors = 166252147)